### PR TITLE
Remove KeytipData from components

### LINKED ISSUE #14674: Remove KeytipData from components and document migration plan
Remove `KeytipData` from components (in favor of `useKeytipData` added in #13688), and add more details to release notes for migration plan.

### PR DESCRIPTION
**Pull request by xugao**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #14674
- [x] Include a change request file using `$ yarn change`

#### Description of changes
- Removed `KeytipData` from components except for legacy buttons and `ContextualMenu` (ones that do not support passing `ref`)
- Updated release notes including migration plan.
- Updated `Keytips.Basic.Examples`

#### Focus areas to test
[Keytips examples on doc site](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/15534/merge/public-docsite-resources/index.html#/examples/keytips)

### CODE REVIEW

File: packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts [hunk 1]
```diff
@@ -125,11 +124,6 @@ export interface ICheckboxProps
    */
   checkmarkIconProps?: IIconProps;
 
-  /**
-   * Optional keytip for this checkbox
-   */
-  keytipProps?: IKeytipProps;
```
Comment (ecraig12345) [inline packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts:131]: Should this (and other removed props) have been marked as deprecated already in 7?

File: packages/react-examples/.storybook/preview.js [hunk 2]
```diff
@@ -30,7 +31,6 @@ if (
 }
 if (['react'].includes('PACKAGE_NAME')) {
```
Comment (ecraig12345) [inline packages/react-examples/.storybook/preview.js:32]: Remove this condition since you're adding the decorator above now?

**Review by ecraig12345 — APPROVED** [review]

File: packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts (hunk 1 above)
Comment (xugao) [reply packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts:131]: Isn't it too late for that ? :/

Comment (ecraig12345) [reply packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts:131]: We could still add it in the 7 branch so there's a bit of warning maybe? Though that assumes they have the deprecation lint rule enabled (and are using things in a way the rule can catch). :(

**Comment by kelseyyoung** [comment]
Just to clarify.....are these only getting deleted in Fabric v8? But will still exist in v7?

**Comment by ecraig12345** [comment]
> Just to clarify.....are these only getting deleted in Fabric v8? But will still exist in v7?

That's correct

**Review by kelseyyoung — APPROVED** [review]

File: packages/react-next/RELEASE_NOTES.md [hunk 3]
```diff
@@ -129,6 +129,34 @@ If you're using a removed `ISpinButtonStyles` prop, you can instead pass a style
 
 Please see the [`@fluentui/react-theme-provider` package README](https://github.com/microsoft/fluentui/blob/master/packages/react-theme-provider/README.md) for details about usage and a migration guide.
 
+### Keytips
+
+Previously, `KeytipData` is built in different components which needs Keytips support. This adds extra bundle size to our components. In version 8, we removed `KeytipData` and `keytipProps` prop from `Link`, `Toggle`, `Checkbox`, `Comboxbox`, `Dropdown`, `SpinButton` and other non-compat `Button`s (the ones which are not exported from `lib/compat`).
```
Comment (khmakoto) [inline packages/react-next/RELEASE_NOTES.md:134]:
```suggestion
Previously, `KeytipData` was built in different components which needed Keytip support. This added extra bundle size to our components. In version 8, we have removed `KeytipData` and `keytipProps` props from `Link`, `Toggle`, `Checkbox`, `ComboBox`, `Dropdown`, `SpinButton` and other non-compat `Button`s (the ones which are not exported from `lib/compat`).
```

**Review by khmakoto — APPROVED** [review]
Approved with one suggestion.

**Pull request merged**